scripts/BLmotors

```diff
@@ -0,0 +1,311 @@
+        res = None
+    f.close()
+    if res is None:
+        return None
+    for lin in res[1]:
+        if 'disable'not in lin.keys():
+            lin['disable'] = False
+        if 'hard'not in lin.keys():
+            lin['hard'] = False
+        if 'history' not in lin.keys():
+            lin['history'] = []
+        if 'alias' not in lin.keys():
+            lin['alias'] = ""
+        if lin['hard']:
+            lin['base'] = ""
+            lin['dir'] = ""
+            lin['host'] = lin['id']
+            lin['port'] = -1
+            lin['rhost'] = lin['id']
+            lin['rport'] = -1
+            lin['rdir'] = lin['dir']
+            lin['newstyle'] = False
+        else:
+            lin['rid'] = lin['id']
+            lin['rdir'] = lin['dir']
+            lin['rhost'] = lin['host']
+            lin['rport'] = lin['port']
+            lin['newstyle'] = False
+    return config
+
+
+def findId(config):
+    """
+    This function takes the list of lists 'procmgr_config'
+    from the tuple 'config' and searchs for the IMS motors.
+    Then stores the id of them into a list.
+    """
+    configFile = config['procmgr_config']
+    listId = []
+    ims = "/ims/"
+    for confile in range(len(configFile)):
+        if (ims in configFile[confile]['dir']) and\
+           (configFile[confile]['disable'] is False):
+            listId.append(configFile[confile]['id'])
+    return listId
+
+
+def assignFullpath(listId):
+    """
+    This function creates a list with the pathname
+    for each ims motor's id from the list 'listId'
+    """
+    listsPath = []
+    for id in range(len(listId)):
+        listsPath.append('/reg/d/iocData/' + listId[id] +
+                         '/iocInfo/IOC.pvlist')
+    return listsPath
+
+
+def createListMotors(listsPath):
+    """
+    This function iterates through the listsPath, opening
+    each IOC.pvlist file to collect all the records (aliases)
+    with the IMS tag. Then it will store them in a list.
+    """
+    motors = []
+    motorsName = []
+    for path in range(len(listsPath)):
+        filename = listsPath[path]
+        with open(filename, 'r') as file:
+            for line in file:
+                comma = line.find(',')
+                if "ims" in line:
+                    motors.append(line[0:comma])
+                    motorsName.append(line[0:comma] + '.NAME')
+    return motors, motorsName
+
+
+def motorsInfo(motors, motorsName, cfg):
+    """
+    This function finds the real Pv name for each
+    record (alias), the cconfiguration name, and if
+    that configuration does matches or ot with the
+    actual configuration. Then it will create a list
+    per each record and finally it will be addes to
+    another list.
+    """
+    pm = pmgrAPI.pmgrAPI("ims_motor", cfg)
+    totalinfo = []
+    for motor in range(len(motors)):
+        infoperRecord = []
+        try:
+            motorPtwo = ophyd.signal.EpicsSignal(motorsName[motor]).get()
```

Comment:
One more thing to note for the future: this line is likely the speed bottleneck for the script. To make it run faster (if that is needed), the EPICS mechanisms would need to be parallelized.